x-pack/plugins/security/server/routes/authentication/common.ts

```diff
@@ -123,66 +132,68 @@ export function defineCommonRoutes({
     return undefined;
   }
 
-  router.post(
-    {
-      path: '/internal/security/login',
-      validate: {
-        body: schema.object({
-          providerType: schema.string(),
-          providerName: schema.string(),
-          currentURL: schema.string(),
-          params: schema.conditional(
-            schema.siblingRef('providerType'),
-            schema.oneOf([
-              schema.literal(BasicAuthenticationProvider.type),
-              schema.literal(TokenAuthenticationProvider.type),
-            ]),
-            basicParamsSchema,
-            schema.never()
-          ),
-        }),
+  if (buildFlavor !== 'serverless') {
```

Comment:
Providing something like this could solve the issue brought up about how many tests are written (logging in as a user with specific privileges). Do you have any ideas how we could do that?